graylog2-server/src/main/java/org/graylog2/indexer/SetIndexReadOnlyJob.java

```diff
@@ -70,9 +77,9 @@ public void execute() {
 
         activityWriter.write(new Activity("Flushed and set <" + index + "> to read-only.", SetIndexReadOnlyJob.class));
 
-        if (!disableIndexOptimization) {
+        if (!indexSet.get().getConfig().indexOptimizationDisabled()) {
             try {
-                systemJobManager.submit(optimizeIndexJobFactory.create(index));
+                systemJobManager.submit(optimizeIndexJobFactory.create(index, indexSet.get().getConfig().indexOptimizationMaxNumSegments()));
```

Comment:
Mostly a nitpick anyway, I've set it to aggressively warn about this (and it most likely doesn't resolve the negations properly).